File: app/src/main/java/com/example/pagomovil/ui/balance/BalanceFragment.java
package com.example.pagomovil.ui.balance;

import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.telephony.SmsManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.example.pagomovil.R;

public class BalanceFragment extends Fragment {

    private BalanceViewModel mViewModel;
    private final String NOMBER = "2662";
    private boolean isBusy = false;

    protected Button csc;
    protected Button cst;
    protected Button cmc;
    protected Button cmt;

    public static BalanceFragment newInstance() {
        return new BalanceFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.balance_fragment, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(BalanceViewModel.class);
        // TODO: Use the ViewModel
    }
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        csc = view.findViewById(R.id.csc);
        cst = view.findViewById(R.id.cst);
        cmc = view.findViewById(R.id.cmc);
        cmt = view.findViewById(R.id.cmt);

        csc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sendMessage("CSC");
            }
        });

        cst.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sendMessage("CST");
            }
        });

        cmc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sendMessage("CMC");
            }
        });

        cmt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sendMessage("CMT");
            }
        });
    }

    protected void sendMessage(String text){
        if (isBusy)
            return;

        isBusy = true;
        try {
            SmsManager sms = SmsManager.getDefault();
            sms.sendTextMessage(NOMBER, null, text, null, null);

            showToast("Procesando...");
        }
        catch (Exception e){
            showToast(e.getMessage());
        }
        finally {
            isBusy = false;
        }
    }

    protected void showToast(String text){
        Toast.makeText(getContext(), text, Toast.LENGTH_LONG).show();
    }
}

File: app/src/main/java/com/example/pagomovil/views/BalanceFragment.kt
package com.example.pagomovil.views

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.Toolbar
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.example.pagomovil.R
import com.example.pagomovil.utilis.app
import com.example.pagomovil.utilis.getViewModel
import com.example.pagomovil.viewModels.BalanceViewModel
import com.example.pagomovil.viewModels.ViewModelComponent
import com.example.pagomovil.viewModels.ViewModelModule

class BalanceFragment : Fragment() {

    private lateinit var viewModelComponent: ViewModelComponent

    private val viewModel: BalanceViewModel by lazy {
        getViewModel { viewModelComponent.balanceViewModel }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModelComponent = requireContext().app.component.inject(ViewModelModule())
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        setupActivity()
        return setupBinding(container)
    }

    private fun setupActivity(){
        //val title = requireActivity().findViewById<Toolbar>(R.id.toolBar)
        //title.title = getString(R.string.consulta)
        val title = requireActivity().findViewById<Toolbar>(R.id.toolBar) ?: return
        title.title = getString(R.string.consulta)
    }

    private fun setupBinding( container: ViewGroup?) : View? {
        val fragmentPaymentBinding: com.example.pagomovil.databinding.FragmentBalanceBinding
                = DataBindingUtil.inflate(layoutInflater, R.layout.fragment_balance, container,false)

        fragmentPaymentBinding.viewModel = this.viewModel
        return  fragmentPaymentBinding.root
    }
}